[msan][NFC] Add tests for Arm NEON smaxv/sminv

thurstond opened the pull request:
This patch precommits tests for the smaxv/sminv intrinsics, which are currently handled suboptimally by visitInstruction.

These are the signed versions of umaxv/uminv (https://github.com/llvm/llvm-project/pull/129661).

Future work will update MSan to apply handleVectorReduceIntrinsic.

thurstond committed:
[msan][NFC] Add tests for Arm NEON smaxv/sminv

This patch precommits tests for the smaxv/sminv intrinsics, which are currently handled suboptimally by visitInstruction.
These are the signed versions of umaxv/uminv (https://github.com/llvm/llvm-project/pull/129661).

Future work will update MSan to apply handleVectorReduceIntrinsic.

thurstond committed:
Fix umax and uminv tests

fmayer commented:
> These are the signed versions of umaxv/uminv (https://github.com/llvm/llvm-project/pull/129661), which are also updated in this patch to add a missing sanitize_memory attribute.

Maybe split into a separate CL?

thurstond committed:
Revert "Fix umax and uminv tests"

This reverts commit 437f872db005592f249131ea8a68980ae032ceff.

thurstond commented:
> > These are the signed versions of umaxv/uminv (#129661), which are also updated in this patch to add a missing sanitize_memory attribute.
> 
> Maybe split into a separate CL?

I've removed them from this pull request https://github.com/llvm/llvm-project/pull/129741/commits/f1e2f2b3d1c3a924a9a68c99882600f00a375c68; separate CL coming soon.
